librz/core/creg.c

```diff
@@ -110,6 +110,26 @@ RZ_API void rz_core_reg_update_flags(RzCore *core) {
 	regs_to_flags(core, rz_core_reg_default(core));
 }
 
+/**
+ * \brief Update registers content of the given type from/to the registers and update/create flags
+ *
+ * Registers are taken from rz_core_reg_default(). Flags in Rizin are updated
+ * according to the current value of the registers.
+ *
+ * \param core RzCore reference
+ * \param type type of register to consider, according to \p RzRegisterType
+ * \param write when true the content of the register is written from \p RzReg
+ * into the debug plugin, if any. Otherwise the content of the register is taken
+ * from the debug plugin, if any, and written into the \p RzReg.
+ */
+RZ_API bool rz_core_reg_update_flags_type(RzCore *core, RzRegisterType type, bool write) {
```

Comment:
This is a VERY strange api. It is mixing up syncing, filtering and flags updating weirdly. I think it will be better to just call `rz_core_reg_update_flags()` at an appropriate place in cutter instead of this.